src/complex/Utilities/DataArrayUtilities.hpp

```diff
@@ -893,4 +895,215 @@ class CopyTupleUsingIndexList
   IDataArray& m_NewCellArray;
   nonstd::span<const int64> m_NewIndices;
 };
+
+/**
+ * @brief The following functions and classes are meant to make copying data from one IArray into another easier for the developer.
+ *
+ * An example use of these functions would be the following (where newCellData is an AttributeMatrix in dataStructure ):
+ *   ParallelTaskAlgorithm taskRunner;
+ *   for (const auto& [dataId, dataObject] : *newCellData)
+ *   {
+ *     auto* inputDataArray = dataStructure.getDataAs<IArray>(inputCellDataPath.createChildPath(name));
+ *     auto* destDataArray = dataStructure.getDataAs<IArray>(destCellDataPath.createChildPath(name));
+ *     auto* newDataArray = dataStructure.getDataAs<IArray>(newCellDataPath.createChildPath(name));
+ *     const IArray::ArrayType arrayType = destDataArray->getArrayType();
+ *     CopyFromArray::RunParallel<CopyFromArray::Combine>(arrayType, destDataArray, taskRunner, inputDataArray, newDataArray);
+ *   }
+ *   taskRunner.wait();
+ * @code
+ *
+ * @endcode
+ */
+namespace CopyFromArray
+{
+/**
+ * @brief Appends all of the data from the inputArray into the destination array starting at the given offset. This function DOES NOT do any bounds checking!
+ */
+template <class K>
+static void AppendData(const K* inputArray, K* destArray, usize offset)
```

Comment:
```suggestion
void AppendData(const K& inputArray, K& destArray, usize offset)
```
I think this is the only other change I would make.